src/filter/owner.rs

```diff
@@ -0,0 +1,113 @@
+use anyhow::{anyhow, Result};
+use std::fs;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct OwnerFilter {
+    uid: Check<u32>,
+    gid: Check<u32>,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+enum Check<T> {
+    Equal(T),
+    NotEq(T),
+    Ignore,
+}
+
+impl OwnerFilter {
+    pub fn from_string(input: &str) -> Result<Self> {
+        let mut it = input.split(':');
+        let (fst, snd) = (it.next(), it.next());
+
+        let uid = Check::parse(fst, |s| {
+            s.parse()
+                .ok()
+                .or_else(|| users::get_user_by_name(s).map(|user| user.uid()))
+                .ok_or_else(|| anyhow!("'{}' is not a recognized user name", s))
+        })?;
+        let gid = Check::parse(snd, |s| {
+            s.parse()
+                .ok()
+                .or_else(|| users::get_group_by_name(s).map(|group| group.gid()))
+                .ok_or_else(|| anyhow!("'{}' is not a recognized group name", s))
+        })?;
+
+        if let (Check::Ignore, Check::Ignore) = (uid, gid) {
```

Comment:
This would only trigger if a user called `--owner :`, right? I'm just a bit irritated by the error message "is not a valid user/group specifier".

Reply:
Yes, that's the only case for which it should trigger.
Do you have any proposals for the error message ?
Maybe something like "At least least a user or group must be specified as owner" ?

Reply:
Maybe we should simply allow it? That would remove some error handling code and I don't see this being used by accident. It might even be useful when someone calls `fd --owner …` from a script.

Hm.. I guess it would also trigger for `--owner ''`? Do we want to allow this as well?

Reply:
FOllowing the script usage scenario, it seems reasonable that when an empty string is passed, `--owner ''` should be a noop as well.

Reply:
I ended up leaving this last condition in, but this time  to make `--owner ''` truly equivalent to not passing `--owner` at all